io/io/inc/ROOT/RFile.hxx

```diff
@@ -231,7 +239,8 @@ class RFile final {
 
    /// Gets object `path` from the file and returns an **owning** pointer to it.
    /// The caller should immediately wrap it into a unique_ptr of the type described by `type`.
-   [[nodiscard]] void *GetUntyped(std::string_view path, const std::type_info &type) const;
+   [[nodiscard]] void *GetUntyped(std::string_view path,
+                                  std::variant<const char *, std::reference_wrapper<const std::type_info>> type) const;
```

Comment:
Since it is private and only use in the source file, would this be an option:
```suggestion
   template <typename T>
   [[nodiscard]] void *GetUntyped(std::string_view path,
                                  T type) const;
```
(`TClass::GetClass` itself has the 2 overloads).